ddtrace/tracer/tracer_test.go

```diff
@@ -1095,6 +1101,20 @@ func decode(p *payload) (spanLists, error) {
 	return traces, err
 }
 
+func (t *dummyTransport) waitFlush(timeout time.Duration) {
+	c := make(chan struct{})
+	go func() {
+		t.Lock()
+		defer t.Unlock()
+		defer close(c)
+		t.sendsig.Wait()
```

Comment:
`sync.Cond` has a lot of bad rep. Honestly I'm not even sure I completely understand it's use, but generally it seems to be used differently (similar to the example given in its godoc). I would like it if we wouldn't use it. Wouldn't it work to do something like:
```
expire := time.After(1*time.Second)
select {
    case <-expire:
        t.Fatal("timed out")
    default:
        if len(t.Traces) == n {
            return
        }
        time.Sleep(5*time.Millisecond)
}
```
Where `n` would be the number of traces to wait for. Then we could have:
```
forceFlush(transport, 4) // flush and wait for 4 traces
```
Or even rename it to `flushAndWait(4)` and get the transport as `t.config.transport.(*dummyTransport)`

Reply:
Following up on my statement about "not using sync.Cond":

> [...] perhaps we should also gently discourage its use.

Ref: https://github.com/golang/go/issues/20491#issuecomment-306996492

See also https://github.com/golang/go/issues/20491#issuecomment-365638697, https://github.com/golang/go/issues/23211#issuecomment-353356407, https://github.com/golang/go/issues/24952#issuecomment-382853850 etc.

Reply:
👍